src/openrct2/world/MapHelpers.cpp

```diff
@@ -381,10 +385,10 @@ sint32 tile_smooth(sint32 x, sint32 y)
     }
 
     // Count number from the three tiles that is currently higher
-    sint8 thresholdW = clamp(neighbourHeightOffset.SW, 0, 1) + clamp(neighbourHeightOffset.W, 0, 1) + clamp(neighbourHeightOffset.NW, 0, 1);
-    sint8 thresholdN = clamp(neighbourHeightOffset.NW, 0, 1) + clamp(neighbourHeightOffset.N, 0, 1) + clamp(neighbourHeightOffset.NE, 0, 1);
-    sint8 thresholdE = clamp(neighbourHeightOffset.NE, 0, 1) + clamp(neighbourHeightOffset.E, 0, 1) + clamp(neighbourHeightOffset.SE, 0, 1);
-    sint8 thresholdS = clamp(neighbourHeightOffset.SE, 0, 1) + clamp(neighbourHeightOffset.S, 0, 1) + clamp(neighbourHeightOffset.SW, 0, 1);
+    sint8 thresholdW = Math::Clamp(neighbourHeightOffset.SW, 0, 1) + Math::Clamp(neighbourHeightOffset.W, 0, 1) + Math::Clamp(neighbourHeightOffset.NW, 0, 1);
```

Comment:
Is this correct? It looks like a bug introduced by thinking the clamp has signature `clamp(value, min, max)`

Reply:
Well, I encountered this earlier. Basically, it will _work_ correctly. It first takes the highest of low and val, then the lowest of max and that value.

Still evil, though.

Reply:
There were many in `console.c` which were wrong. We should make both of our clamp signatures `(value, min, max)` which conforms with `std::clamp`.